lib/ansible/plugins/action/__init__.py

```diff
@@ -1039,9 +1039,11 @@ def _low_level_execute_command(self, cmd, sudoable=True, in_data=None, executabl
             display.debug("_low_level_execute_command(): changing cwd to %s for this command" % chdir)
             cmd = self._connection._shell.append_command('cd %s' % chdir, cmd)
 
-        if (sudoable and self._connection.transport != 'network_cli' and self._connection.become and
-                (C.BECOME_ALLOW_SAME_USER or
-                 self.get_become_option('become_user') != self._get_remote_user())):
+        ruser = self._get_remote_user()
+        buser = self.get_become_option('become_user')
+        if (sudoable and self._connection.become and  # if sudoable and have become
```

Comment:
This conditional is complex enough that I think it would be helpful to have the logic broken out and in its own function.
```suggestion
        if self._should_i_become(sudoable, buser, ruser):
```

```python
def _should_i_become(self, sudoable, buser, ruser):
    # Should we become?
    if not sudoable:
        return False

    # Does the connection support become?
    if not self._connection.become or self._connection.transport == 'network_cli':
        return False

    # Corner case of trying to become the same user
    if ruser or buser and (buser != ruser):
        # Only allow becoming same user if config says so
        if not C.BECOME_ALLOW_SAME_USER:
            return False

    return True
```

Reply:
at 3 conditions that are not reused, i still think it doesn't merit it's own function.

Reply:
It could still be broken out without being a separate function. Having a separate function reads easier, but I am mostly suggesting the logic be broken out more for easier reading and maintenance.

Reply:
i would prefer not to refactor at the same time as i change the condition ... tends to lead to more changes in behaviour than desired.